pandas/core/strings.py

```diff
@@ -323,15 +328,24 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
+    case : boolean, default None
```

Comment:
another possiblity here is to make ``case=object`` the default, this allows you to detect if anything is passed, with the nice propery that ``object`` is ``True`` (while ``None`` is ``False``). Its the same idea, but less error prone.